test/LocatedTriplesTest.cpp

```diff
@@ -494,6 +492,114 @@ TEST_F(LocatedTriplesTest, mergeTriples) {
   }
 }
 
+// Test the `mergeTriples` functions with inputs that contain different graphs.
+TEST_F(LocatedTriplesTest, mergeTriplesWithGraph) {
+  using LT = LocatedTriple;
+  std::vector<CompressedBlockMetadata> emptyMetadata;
+
+  // Merge the `LocatesTriples` into a block with 3 index columns.
+  {
+    IdTable block = makeIdTableFromVector({
+        {1, 10, 10, 0},  // Row 0
+        {2, 15, 20, 0},  // Row 1
+        {2, 15, 20, 1},  // Row 2
+        {2, 15, 20, 2},  // Row 3
+        {2, 15, 30, 1},  // Row 4
+        {2, 15, 30, 3},  // Row 5
+        {3, 30, 30, 0}   // Row 6
+    });
+    auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock({
+        LT{1, IT(1, 5, 10, 3), true},    // Insert before row 0
+        LT{1, IT(2, 15, 20, 1), false},  // Delete row 2
+        LT{1, IT(2, 15, 20, 3), false},  // Delete non-existent row
+        LT{1, IT(2, 15, 30, 2), true},   //  Insert between 4 and 5
+
+    });
+    IdTable resultExpected = makeIdTableFromVector({
+        {1, 5, 10, 3},  // LT 0
+        {1, 10, 10, 0},
+        {2, 15, 20, 0},  // Row 1
+        {2, 15, 20, 2},  // Row 3
+        {2, 15, 30, 1},  // Row 4
+        {2, 15, 30, 2},  // LT 3
+        {2, 15, 30, 3},  // Row 5
+        {3, 30, 30, 0}   // Row 6
+    });
+
+    auto merged = locatedTriplesPerBlock.mergeTriples(1, block, 3, true);
+    EXPECT_THAT(merged, testing::ElementsAreArray(resultExpected));
+  }
+
+  // Merge the `LocatesTriples` into a block with 2 index columns. This may
+  // happen if all triples in a block have the same value for the first column.
+  {
+    IdTable block = makeIdTableFromVector({
+        {10, 10, 1},  // Row 0
+        {15, 20, 2},  // Row 1
+        {15, 30, 1},  // Row 2
+        {20, 10, 2},  // Row 3
+    });
+    auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock({
+        LT{1, IT(1, 10, 10, 1), false},  // Delete row 0
+        LT{1, IT(1, 13, 20, 3), true},   // Insert before row 1
+        LT{1, IT(1, 15, 20, 1), true},   // Insert before row 1
+        LT{1, IT(1, 15, 20, 2), true},   // Insert already existing row
+        LT{1, IT(1, 20, 10, 1), false},  // Delete non-existent row
+    });
+
+    IdTable resultExpected = makeIdTableFromVector({{13, 20, 3},    // LT 1
+                                                    {15, 20, 1},    // LT 2
+                                                    {15, 20, 2},    // Row 1
+                                                    {15, 30, 1},    // Row 2
+                                                    {20, 10, 2}});  // Row 3
+
+    auto merged = locatedTriplesPerBlock.mergeTriples(1, block, 2, true);
+    EXPECT_THAT(merged, testing::ElementsAreArray(resultExpected));
```

Comment:
How about also testing here (and in the following) the call, where the last argument is `false`. A simple helper lambda that strips the last column from `resultExpected` should do the job (and it can be reused for the following cases).

Reply:
If I understand you correctly, then this doesn't work. The result changes as soon as we drop the graph column, as triples that don't match because of the GRAPH now suddenly match etc.
We already (preexisting, so not shown in the diff) have exhaustive tests for the case where the last argument is `false`.

Reply:
You are right, thanks for the clarification